cli/js/os.ts

```diff
@@ -133,13 +133,247 @@ export function start(preserveDenoNamespace = true, source?: string): Start {
  * Requires the `--allow-env` flag.
  */
 export function homeDir(): string {
-  const path = sendSync(dispatch.OP_HOME_DIR);
+  const path = sendSync(dispatch.OP_GET_DIR, { name: "home" });
   if (!path) {
     throw new Error("Could not get home directory.");
   }
   return path;
 }
 
+/**
+ * Returns the current user's cache directory.
+ * If the directory does not exist, an exception is thrown
+ * Requires the `--allow-env` flag.
+ * |Platform | Value                               | Example                      |
+ * | ------- | ----------------------------------- | ---------------------------- |
+ * | Linux   | `$XDG_CACHE_HOME` or `$HOME`/.cache | /home/alice/.cache           |
+ * | macOS   | `$HOME`/Library/Caches              | /Users/Alice/Library/Caches  |
+ * | Windows | `{FOLDERID_LocalAppData}`           | C:\Users\Alice\AppData\Local |
+ */
+export function cacheDir(): string {
+  const path = sendSync(dispatch.OP_GET_DIR, { name: "cache" });
+  if (!path) {
+    throw new Error("Could not get user cache directory.");
+  }
```

Comment:
All of these places where you throw new Error can be removed if you return the error from the op.

Reply:
done